src/benchmarklib/benchmark_runner.cpp

```diff
@@ -349,13 +356,16 @@ std::vector<std::shared_ptr<AbstractTask>> BenchmarkRunner::_schedule_query(
     query_tasks.insert(query_tasks.end(), tasks.begin(), tasks.end());
   }
 
+  *metrics = pipeline.metrics();
```

Comment:
On mobile, so I'm probably missing something. Why do we need to dereference metrics?

Reply:
to assign by value. could assign by shared_ptr probably wouldn't make a difference.

Reply:
removed